### ISSUE TITLE
Exif info not available

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#246]
**[Original report](https://bitbucket.org/agriggio/art/issues/246) by Paul Matthijsse (Bitbucket: [Paul Matthijsse](https://bitbucket.org/Paul Matthijsse), ).**

----------------------------------------

Hello, there’s a little problem with the exif info \(shift\+I\), it says not available, although it is of course, see screenshot. **Some** thumbnails only show the file name, others file name \+ exif. Opening one of the last in the editor gives the same: exif info not available.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Hi,

please provide a sample picture, thanks.

**Comment by agriggio** [comment]
**Original comment by Paul Matthijsse (Bitbucket: [Paul Matthijsse](https://bitbucket.org/Paul Matthijsse), ).**

----------------------------------------

Sorry, can’t upload raw files here and they must be smaller than 1MB. 

So I thought to resize the raw and save it as jpg and upload that one, but there the exif is shown as usual…

I’ll recompile ART to see if it gets back to its normal behaviour.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

hi,

You can also use the forum in this case. Without a way to reproduce, there's not much I can do unfortunately…

**Comment by agriggio** [comment]
**Original comment by Paul Matthijsse (Bitbucket: [Paul Matthijsse](https://bitbucket.org/Paul Matthijsse), ).**

----------------------------------------

Thought about the forum as well, but I’ll first rebuild ART.

**Comment by agriggio** [comment]
**Original comment by Paul Matthijsse (Bitbucket: [Paul Matthijsse](https://bitbucket.org/Paul Matthijsse), ).**

----------------------------------------

Hello, see forum.

**Comment by agriggio** [comment]
**Original comment by t.guillermin (Bitbucket: [t.guillermin](https://bitbucket.org/t.guillermin), ).**

----------------------------------------

Hello

I have the same problem,.

when I use Verbose=1 in option,  I get this message:

EXIV2 ERROR: Invalid tag name or ifdId \`RFLensType', ifdId 28

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

thanks for the info. What version of exiv2 are you using?

**Comment by agriggio** [comment]
**Original comment by t.guillermin (Bitbucket: [t.guillermin](https://bitbucket.org/t.guillermin), ).**

----------------------------------------

Yes.

I found this in the EXIV2 documentation:

Constructor to create an Exif key from a key string.

Parameters

**Exceptions**

Error: if the first part of the key is not '**Exif**' or the remainin parts of the key cannot be parsed and converted to a group name and tag name.

In the rtengine/imagedata.cc file, I made this change:

```
  const auto find_exif_tag =
        [&](const std::string &name) -> bool
        {
            try {
                pos = exif.findKey(Exiv2::ExifKey(name));
                return (pos != exif.end() && pos->size());                
            } catch (std::exception &e) {
                if (settings->verbose) {                        
                    std::cerr << "find_exif_tag(" << name << "): " << e.what() << std::endl;
                }
                return false;
            }                
        };
```

And after deleting the ART directory from the linux cache, it's OK.

‌

**Comment by agriggio** [comment]
**Original comment by t.guillermin (Bitbucket: [t.guillermin](https://bitbucket.org/t.guillermin), ).**

----------------------------------------

Hi,  
I updated the exiv2 with source file version 0.27.5 \(exiv2-0.27.5-Source.tar.gz\), RFLensType is now set.  
Before, I tried by downloading the source from github but the toLong function of the Value class \(value.hpp\) was replaced by toUint32. This causes an ART compilation error.

Thierry.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Thanks for the patch!

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

metadata: fixed regression with older exiv2 versions

Fixes #246

**Comment by agriggio** [comment]
**Original comment by Paul Matthijsse (Bitbucket: [Paul Matthijsse](https://bitbucket.org/Paul Matthijsse), ).**

----------------------------------------

Works again now. Thanks Thierry for your patch!
